rustls-cert-gen/src/cert.rs

```diff
@@ -0,0 +1,433 @@
+use rcgen::{
+	BasicConstraints, Certificate, CertificateParams, DistinguishedName, DnType,
+	DnValue::PrintableString, ExtendedKeyUsagePurpose, IsCa, KeyUsagePurpose, SanType,
+};
+use std::{fs::File, io, path::Path};
+
+#[derive(Debug, Clone)]
+/// Pem serialized Certificate and Pem serialized corresponding private key
+pub struct PemCertifiedKey {
+	pub cert_pem: String,
+	pub private_key_pem: String,
+}
+
+impl PemCertifiedKey {
+	pub fn write(&self, dir: &Path, name: &str) -> Result<(), io::Error> {
+		use std::io::Write;
+		std::fs::create_dir_all(dir)?;
+
+		let key_path = dir.join(format!("{name}.key.pem"));
+		let mut key_out = File::create(key_path)?;
+		write!(key_out, "{}", &self.private_key_pem)?;
+
+		let cert_path = dir.join(format!("{name}.pem"));
+		let mut cert_out = File::create(cert_path)?;
+		write!(cert_out, "{}", &self.cert_pem)?;
+
+		Ok(())
+	}
+}
+
+/// Builder to configure TLS [CertificateParams] to be finalized
+/// into either a Ca or an End-Entity.
```

Comment:
```suggestion
/// into either a [Ca] or an [EndEntity].
```

Reply:
This change results in warnings from  `cargo doc`  causing some CI jobs to fail. The warnings are:
```
warning: public documentation for `CertificateBuilder` links to private item `EndEntity`
```
Something must be escaping my attention, but I don't quite understand the warning.

Reply:
I think this is related to [feedback I left on the other PR](https://github.com/rustls/rcgen/pull/185#discussion_r1385550049).

If the intent is for this code to be usable as a library in addition to being useful through the CLI then I think in `rustls-cert-gen/src/lib.rs` where you have `pub use cert::CertificateBuilder` you should add the CA and EE types as `pub` exports.

Reply:
Oh yes I forgot about that `lib.rs`.